models/migrations/v18.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/go-xorm/xorm"
+	"github.com/gogits/gogs/pkg/setting"
+)
+
+func updateRepositoryDescriptionField(x *xorm.Engine) error {
+	exist, err := x.IsTableExist("repository")
+	if err != nil {
+		return fmt.Errorf("IsTableExist: %v", err)
+	} else if !exist {
+		return nil
+	}
+	if (setting.UseMySQL) {
+		_, err = x.Exec("ALTER TABLE `repository` MODIFY `description` TEXT")
```

Comment:
Why not change to `VARCHAR(512)`?

Reply:
Changed to `VARCHAR(512)`.
`TEXT` used in previous version for `4000` long string.